src/admin.c

```diff
@@ -1330,6 +1330,79 @@ static bool admin_cmd_enable(PgSocket *admin, const char *arg)
 	return admin_ready(admin, "ENABLE");
 }
 
+
+
+static PgSocket *find_socket_in_list(PgSocket *target_client, struct StatList *sockets)
+{
+	struct List *item;
+	PgSocket *socket;
+
+	statlist_for_each(item, sockets) {
+		socket = container_of(item, PgSocket, head);
+		if (target_client == socket) {
+			return socket;
+		}
+	}
+	return NULL;
+}
+
+static PgSocket *find_client_global(PgSocket *target_client)
+{
+	PgSocket *kill_client;
+	struct List *item;
+	PgPool *pool;
+
+	statlist_for_each(item, &pool_list) {
+		pool = container_of(item, PgPool, head);
+
+		kill_client = find_socket_in_list(target_client, &pool->active_client_list);
+		if (kill_client != NULL) {
+			return kill_client;
+		}
+		kill_client = find_socket_in_list(target_client, &pool->waiting_client_list);
+		if (kill_client != NULL) {
+			return kill_client;
+		}
+		kill_client = find_socket_in_list(target_client, &pool->active_cancel_req_list);
+		if (kill_client != NULL) {
+			return kill_client;
+		}
+		kill_client = find_socket_in_list(target_client, &pool->waiting_cancel_req_list);
+		if (kill_client != NULL) {
+			return kill_client;
+		}
+	}
+
+	statlist_for_each(item, &peer_pool_list) {
+		pool = container_of(item, PgPool, head);
+
+		kill_client = find_socket_in_list(target_client, &pool->active_cancel_req_list);
+		if (kill_client != NULL) {
+			return kill_client;
+		}
+		kill_client = find_socket_in_list(target_client, &pool->waiting_cancel_req_list);
+		if (kill_client != NULL) {
+			return kill_client;
+		}
+	}
+	return NULL;
+}
+
+/* Command: KILL_CLIENT */
+static bool admin_cmd_kill_client(PgSocket *admin, const char *arg)
+{
+	PgSocket *kill_client;
+	PgSocket *target_client = NULL;
+
+	sscanf(arg, "%p", &target_client);
```

Comment:
Let's check the return value of `sscanf` and differentiate in the returned error message when the user uses an incorrect formatted value vs an unknown pointer value.

Reply:
Addressed in last commit. Added  a unit test for this as well so now there are 3 unit tests associated with this PR.